R/flea_starters.R

```diff
@@ -27,44 +27,56 @@ ff_starters.flea_conn <- function(conn, week = 1:17, ...) {
 }
 
 .flea_starters <- function(week, game_id, conn) {
+  cols <- c(injurytypeAbbreviaition = NA, injurytypeFull = NA, injurydescription = NA, injuryseverity = NA)
   x <- fleaflicker_getendpoint("FetchLeagueBoxscore",
-    sport = "NFL",
-    scoring_period = week,
-    fantasy_game_id = game_id,
-    league_id = conn$league_id
-  ) %>%
-    purrr::pluck("content", "lineups") %>%
-    list() %>%
-    tibble::tibble() %>%
-    tidyr::unnest_longer(1) %>%
-    tidyr::unnest_wider(1) %>%
-    tidyr::unnest_longer("slots") %>%
-    tidyr::unnest_wider("slots") %>%
+                               sport = "NFL",
+                               scoring_period = week,
+                               fantasy_game_id = game_id,
+                               league_id = conn$league_id
+  ) |>
+    purrr::pluck("content", "lineups") |>
+    list() |>
+    tibble::tibble() |>
+    tidyr::unnest_longer(1) |>
+    tidyr::unnest_wider(1) |>
+    tidyr::unnest_longer("slots") |>
+    tidyr::unnest_wider("slots") |>
     dplyr::mutate(
       position = purrr::map_chr(.data$position, purrr::pluck, "label"),
       positionColor = NULL
-    ) %>%
-    tidyr::pivot_longer(c("home", "away"), names_to = "franchise", values_to = "player") %>%
-    tidyr::hoist("player", "proPlayer", "owner", "points" = "viewingActualPoints") %>%
+    ) |>
+    tidyr::pivot_longer(c("home", "away"), names_to = "franchise", values_to = "player") |>
+    tidyr::hoist("player", "proPlayer", "owner", "points" = "viewingActualPoints") |>
     tidyr::hoist("proPlayer",
-      "player_id" = "id",
-      "player_name" = "nameFull",
-      "pos" = "position",
-      "team" = "proTeamAbbreviation"
-    ) %>%
-    dplyr::filter(!is.na(.data$player_id)) %>%
-    tidyr::hoist("owner", "franchise_id" = "id", "franchise_name" = "name") %>%
-    tidyr::hoist("points", "player_score" = "value") %>%
+                 "player_id" = "id",
+                 "player_name" = "nameFull",
+                 "pos" = "position",
+                 "team" = "proTeamAbbreviation",
+                 "injury" = "injury"
+    ) |>
+    # dplyr::filter(!is.na(.data$player_id)) |>
+    tidyr::unnest_wider("injury", names_sep = "") |>
+    tidyr::hoist("owner", "franchise_id" = "id", "franchise_name" = "name") |>
+    tidyr::hoist("points", "player_score" = "value") |>
+    dplyr::mutate(group = dplyr::case_when(is.na(group) == TRUE ~ "BENCH", .default = group))
+  x = tibble::add_column(x, !!!cols[setdiff(names(cols), names(x))]) |>
```

Comment:
using the bind_rows suggestion also avoids this gnarly rlang injection
```suggestion
    dplyr::mutate(group = dplyr::case_when(is.na(group) == TRUE ~ "BENCH", .default = group)) %>%
```